src/main/java/org/elasticsearch/index/query/BaseQueryBuilder.java

```diff
@@ -68,5 +69,14 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
```

Comment:
Right I got confused with the parser, sorry! Shall we introduce these new methods through a new intermediate base class (BaseQueryBuilderTemp?) instead and have a new one for the migrated builders too? maybe overkill... what do you think?